File: celsius_conversion/scala_celsius_conversion/celsius_conversion.scala
object celsius_conversion {
  def example_function( n:Double ) : Double = {
    return n
  }

  def toCelsius( x:Double ) : Double = {
    return (5.0/9.0) * (x - 32.0)
  }

  def toFahrenheit( x:Double ) : Double = {
    return x * (9.0/5.0) + 32.0
  }

  def main(args: Array[String]): Unit = {
    println( "toFahrenheit(21.0)=<" + toFahrenheit(21.0) + ">")
    println( "toCelsius(70.0)=<" + toCelsius(70.0) + ">")
  }
}

File: celsius_conversion/csharp_celsius_conversion/celsius_conversion.cs
using System;

public class CelsiusConversion {
  public static void Main( string[] args ) {
    if( args.Length != 0 ) {
      Console.Error.WriteLine( "Usage: " + Environment.GetCommandLineArgs()[0] + " <noargs>" );
      Environment.Exit(1);
    }

    Console.WriteLine("toFahrenheit(" + 21.0 + ")=<" + toFahrenheit(21.0) + ">");
    Console.WriteLine("toCelsius(" + 70.0 + ")=<" + toCelsius(70.0) + ">");
  }

  public static double toFahrenheit ( double x ) {
    return  x * (9.0/5.0) + 32.0;
  }

  public static double toCelsius ( double x ) {
    return (x - 32.0) * (5.0/9.0);
  }
}
